client/src/partials/cash/cash.js

```diff
@@ -151,55 +177,29 @@ function CashController(Cash, Cashboxes, AppCache, Currencies, Modal, $statePara
     vm.patient = patient;
   }
 
-  /**
-   * Cash Receipt Modal
-   */
+  /** Receipt Modal */
   function openReceiptModal(uuid) {
-
-    var instance = Modal.open({
-      templateUrl: 'partials/cash/modals/receipt.modal.html',
-      controller:  'CashReceiptModalController as CashReceiptModalCtrl',
-      size:        'md',
-      backdrop:    'static',
-      animation:   false,
-      resolve : {
-        uuid : function uuidProvider() { return uuid; },
-        patientUuid : function patientUuidProvider() { return vm.patient.uuid; }
-      }
-    });
+    ModalService.openPatientReceipt({ uuid: uuid, patientUuid: vm.patient.uuid });
   }
 
-  /**
-   * receive open invoices from the invoice modal
-   *
-   * @todo refactor this into a separate invoices modal component, that can
-   * be injected into any controller.
-   */
-  function openInvoicesModal() {
+  /** Transfer Modal */
+  function openTransferModal() {
+    ModalService.openTransfer({ cashbox: vm.cashbox });
+  }
 
-    var instance = Modal.open({
-      templateUrl: 'partials/cash/modals/invoices.modal.html',
-      controller:  'CashInvoiceModalController as CashInvoiceModalCtrl',
-      size:        'md',
-      backdrop:    'static',
-      animation:   false,
-      resolve:     {
-        debtorId:  function debtorIdProvider() { return vm.payment.debtor_uuid; },
-        invoiceIds : function invoiceIdsProvider() {
-
-          // if no invoices have been initialized, pass in an empty array.
-          if (!vm.payment.invoices) { return []; }
-
-          return vm.payment.invoices.map(function (invoice) {
-            return invoice.sale_uuid;
-          });
-        }
-      }
+  /** Select Cashbox Modal */
+  function openSelectCashboxModal() {
+    ModalService.openSelectCashbox({ cashbox: cache.cashbox, cashboxId: cashboxId })
+    .then(function (result) {
+      vm.cashbox = result;
+      $location.url('/cash/' + vm.cashbox.id);
```

Comment:
Instead of using `$location`, it might be better to use `$state`.  What do you think?